cmd/bootstrap/cmd/access_keygen.go

```diff
@@ -0,0 +1,136 @@
+
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/model/bootstrap"
+	"github.com/onflow/flow-go/utils/grpcutils"
+)
+
+const certValidityPeriod = 100 * 365 * 24 * time.Hour // ~100 years
+
+var (
+	flagSANs           string
+	flagCommonName     string
+	flagNodeInfoFile   string
+	flagOutputKeyFile  string
+	flagOutputCertFile string
+)
+
+var accessKeyCmd = &cobra.Command{
+	Use:   "access-keygen",
+	Short: "Generate access node grpc TLS key and certificate",
+	Run:   accessKeyCmdRun,
+}
+
+func init() {
+	rootCmd.AddCommand(accessKeyCmd)
+
+	accessKeyCmd.Flags().StringVar(&flagNodeInfoFile, "node-info", "", "path to node's node-info.priv.json file")
+	_ = accessKeyCmd.MarkFlagRequired("node-info")
+
+	accessKeyCmd.Flags().StringVar(&flagOutputKeyFile, "key", "./access-tls.key", "path to output private key file")
+	accessKeyCmd.Flags().StringVar(&flagOutputCertFile, "cert", "./access-tls.crt", "path to output certificate file")
+	accessKeyCmd.Flags().StringVar(&flagCommonName, "cn", "", "common name to include in the certificate")
+	accessKeyCmd.Flags().StringVar(&flagSANs, "sans", "", "subject alternative names to include in the certificate, comma separated")
+}
+
+// accessKeyCmdRun generate an Access node TLS key and certificate
+func accessKeyCmdRun(_ *cobra.Command, _ []string) {
+	networkKey, err := loadNetworkKey(flagNodeInfoFile)
+	if err != nil {
+		log.Fatal().Msgf("could not load node-info file: %v", err)
+	}
+
+	certTmpl, err := defaultCertTemplate()
+	if err != nil {
+		log.Fatal().Msgf("could not create certificate template: %v", err)
+	}
+
+	if flagCommonName != "" {
+		log.Info().Msgf("using cn: %s", flagCommonName)
+		certTmpl.Subject.CommonName = flagCommonName
+	}
+
+	if flagSANs != "" {
+		log.Info().Msgf("using SANs: %s", flagSANs)
+		certTmpl.DNSNames = strings.Split(flagSANs, ",")
+	}
+
+	cert, err := grpcutils.X509Certificate(networkKey, grpcutils.WithCertTemplate(certTmpl))
+	if err != nil {
+		log.Fatal().Msgf("could not generate key pair: %v", err)
+	}
+
+	// write cert and private key to disk
+	keyBytes, err := x509.MarshalECPrivateKey(cert.PrivateKey.(*ecdsa.PrivateKey))
+	if err != nil {
+		log.Fatal().Msgf("could not encode private key: %v", err)
+	}
+
+	err = os.WriteFile(flagOutputKeyFile, pem.EncodeToMemory(&pem.Block{
+		Type:  "EC PRIVATE KEY",
+		Bytes: keyBytes,
+	}), 0600)
+	if err != nil {
+		log.Fatal().Msgf("could not write private key: %v", err)
+	}
+
+	err = os.WriteFile(flagOutputCertFile, pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: cert.Certificate[0],
+	}), 0600)
+	if err != nil {
+		log.Fatal().Msgf("could not write certificate: %v", err)
+	}
+}
+
+func loadNetworkKey(nodeInfoPath string) (crypto.PrivateKey, error) {
+	data, err := os.ReadFile(nodeInfoPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not read private node info (path=%s): %w", nodeInfoPath, err)
+	}
+
+	var info bootstrap.NodeInfoPriv
+	err = json.Unmarshal(data, &info)
```

Comment:
I'm a bit confused about the networking key curves in the protocol: 
- the function in this PR assumes there are on P-256 because I think it uses this [unmarshaling](https://github.com/onflow/flow-go/blob/master/model/encodable/keys.go#L92-L103). Are all access nodes keys on P-256? I think I remember a difference between staked and unstaked nodes but I may be wrong.
- Observer nodes [seem to use](https://github.com/onflow/flow-go/blob/master/cmd/observer/node_builder/observer_builder.go#L1069) secp256k1 (btw the function code and purpose seem very similar to what is in the PR).

Reply:
This uses the same logic ANs use to load the networking key:
https://github.com/onflow/flow-go/blob/master/cmd/scaffold.go#L1832

This is only intended for staked Access nodes since observers are untrusted and don't necessarily have a static network identity.

Reply:
Yes,  unstaked access nodes use a secp256k1 networking key: https://github.com/onflow/flow-go/blob/master/cmd/bootstrap/utils/key_generation.go#L52-L53 (unlike staked nodes)